commit 3e5bffa5371d22842c135608605e3bb4f55ffbf0
Author: Lior David <liord@codeaurora.org>
Date:   Wed Oct 31 10:52:14 2018 +0200

    wil6210: fix memory leak in wil_find_tx_bcast_2
    
    [ Upstream commit 664497400c89a4d40aee51bcf48bbd2e4dc71104 ]
    
    A successful call to wil_tx_ring takes skb reference so
    it will only be freed in wil_tx_complete. Consume the skb
    in wil_find_tx_bcast_2 to prevent memory leak.
    
    Signed-off-by: Lior David <liord@codeaurora.org>
    Signed-off-by: Maya Erez <merez@codeaurora.org>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/net/wireless/ath/wil6210/txrx.c b/drivers/net/wireless/ath/wil6210/txrx.c
index 6a7943e487fb..75c8aa297107 100644
--- a/drivers/net/wireless/ath/wil6210/txrx.c
+++ b/drivers/net/wireless/ath/wil6210/txrx.c
@@ -1313,6 +1313,8 @@ static struct wil_ring *wil_find_tx_bcast_2(struct wil6210_priv *wil,
 			wil_dbg_txrx(wil, "BCAST DUP -> ring %d\n", i);
 			wil_set_da_for_vring(wil, skb2, i);
 			wil_tx_ring(wil, vif, v2, skb2);
+			/* successful call to wil_tx_ring takes skb2 ref */
+			dev_kfree_skb_any(skb2);
 		} else {
 			wil_err(wil, "skb_copy failed\n");
 		}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1012_3e5bffa5371d22842c135608605e3bb4f55ffbf0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1012_3e5bffa5371d22842c135608605e3bb4f55ffbf0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
